internal/gatewayapi/contexts.go

```diff
@@ -17,16 +17,18 @@ import (
 type GatewayContext struct {
 	*v1beta1.Gateway
 
-	listeners map[v1beta1.SectionName]*ListenerContext
+	listeners []*ListenerContext
 }
 
 func (g *GatewayContext) GetListenerContext(listenerName v1beta1.SectionName) *ListenerContext {
 	if g.listeners == nil {
-		g.listeners = make(map[v1beta1.SectionName]*ListenerContext)
+		g.listeners = make([]*ListenerContext, 0)
 	}
 
-	if ctx := g.listeners[listenerName]; ctx != nil {
-		return ctx
+	for _, l := range g.listeners {
```

Comment:
if we take this approach of implicitly relying on input listener order, we'll no longer need to sort the listener Names in the o/p, can you also rm https://github.com/envoyproxy/gateway/blob/34afca0f0ad17e4caed3cda7a1ac2a1a1658ab11/internal/gatewayapi/sort.go#L48